app/javascript/packs/clipboardCopy.js

```diff
@@ -1,5 +1,22 @@
-document.addEventListener('DOMContentLoaded', () => {
-  window.WebComponents.waitFor(() => {
-    import('@github/clipboard-copy-element');
-  });
+HTMLDocument.prototype.ready = new Promise(resolve => {
+  if (document.readyState !== 'loading') {
+    return resolve();
+  }
+  document.addEventListener('DOMContentLoaded', () => resolve());
+  return null;
+});
+
+document.ready.then(() => {
+  if (
+    !window.clipboard &&
+    !window.Clipboard &&
+    !navigator.clipboard &&
+    !navigator.Clipboard
+  ) {
+    window.clipboard = import('clipboard-polyfill');
```

Comment:
[dynamic imports](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import) return a promise, so the code as is won't work. `window.clipboard` will be set to a Promise.

If you make the anonymous function `async`, then you can `await`. An alternative would be to do a Promise -> then.

```suggestion
    window.clipboard = await import('clipboard-polyfill');
```

Also consider managing if it fails to load.

Reply:
Oh I did not know that! Thanks!

Reply:
Actually I'm getting the following after adding this
```
ERROR in ./app/javascript/packs/clipboardCopy.js
Module build failed: SyntaxError: await is a reserved word (16:23)

  14 |     !navigator.Clipboard
  15 |   ) {
> 16 |     window.clipboard = await import('clipboard-polyfill');
     |                        ^
  17 |   }
  18 | });
  19 |

BabelLoaderError: SyntaxError: await is a reserved word (16:23)

  14 |     !navigator.Clipboard
  15 |   ) {
> 16 |     window.clipboard = await import('clipboard-polyfill');
     |                        ^
  17 |   }
  18 | });
  19 |
```

Reply:
Hmm, did you add the `async` keyword before the anonymous function?

```javascript
document.ready.then(async () => {
  if (
    !window.clipboard &&
    !window.Clipboard &&
    !navigator.clipboard &&
    !navigator.Clipboard
  ) {
    window.clipboard = await import('clipboard-polyfill');
    });
  }
});
```

If you're still having trouble, I can look at this later tonight.

Reply:
Ah, I did not that's probably why. I did it differently now but I think it will work in the same manner.